src/main/java/kmql/QueryCompleter.java

```diff
@@ -0,0 +1,61 @@
+package kmql;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.jline.reader.Candidate;
+import org.jline.reader.Completer;
+import org.jline.reader.LineReader;
+import org.jline.reader.ParsedLine;
+import org.jline.reader.impl.completer.StringsCompleter;
+
+/**
+ * A {@link Completer} implementation for interactive kmql session.
+ */
+public class QueryCompleter implements Completer {
```

Comment:
Any points to make this a `Completer`? If this class's reponsibility is just to construct a `StringCompleter` along with valid candidates, then would a static factory method that creates a `StringCompleter`  just enough?

Reply:
Oh, sounds good.